### PR TITLE
66 refactor init db for less connections

### PR DESCRIPTION
**Pull request by yx2512**
Fixed:

+ Removed ```db.create_all``` and ```db.drop_all()``` from product.py and wishlist.py

+ Removed ```db.init_app()``` from product.py and wishlist.py

### CODE REVIEW

**Comment by yx2512** [comment]
Our existing code calls ```db.init_app()``` in each model, which leads to multiple database connections. Given the facts the we only have 5 connects to ElephantSQL and we need to have at least two instances running on cloud, I refactored our database init logic so that it only creates one connection per instance but still gets the work done.

**Comment by yx2512** [comment]
We have redundant ```db.drop_all()``` and ```db.create_all()``` calls in each model. This does not make too much sense when the service starts because, once we create all the tables, we don't want them to be discarded just because of another ```init_db()``` call.

File: tests/test_routes.py [hunk 1]
```diff
@@ -57,8 +56,9 @@ def setUpClass(cls):
     app.config["DEBUG"] = False
     app.config["SQLALCHEMY_DATABASE_URI"] = DATABASE_URI
     app.logger.setLevel(logging.CRITICAL)
-    Product.init_db(app)
-    Wishlist.init_db(app)
+    with app.app_context():
+      db.drop_all()
+      db.create_all()
```
Comment (ntlz) [inline tests/test_routes.py:61]: Should this include `Product.init_db(app)` and `Wishlist.init_db(app)` like the other two?

File: service/models/wishlist.py [hunk 2]
```diff
@@ -109,11 +109,6 @@ def init_db(cls, app:Flask):
     logger.info("Wishlist: Initializing database")
     cls.app = app
```
Comment (ntlz) [inline service/models/wishlist.py:110]: Do we even need this method at all? I don't see `cls.app` or `self.app` being referenced anywhere in the class. Or am I missing something on how this is used?

File: service/models/product.py [hunk 3]
```diff
@@ -106,11 +106,6 @@ def init_db(cls, app:Flask):
     logger.info("Product: initializing database")
```
Comment (ntlz) [inline service/models/product.py:106]: Same comment as below (sorry they are out of order 😆 )

File: service/models/wishlist.py (hunk 2 above)
Comment (yx2512) [reply service/models/wishlist.py:110]: Good point. I'm actually not sure about this. But, if you refer to the code at line 25, ```app``` is initialized to be ```None```. I think what I have here is to assign an app instance to this field. Depending on we are doing testing or receiving requests from user, ```app``` contains different config information which might be useful when Flask-SQLAlchemy performs query?

**Review by XinyuC97 — APPROVED** [review]

**Comment by yx2512** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/27b643c6-9a70-4eea-abde-faef8f7042e9?env_id=ibm:yp:us-south "nyu-wishlist-service-fall2103"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/659061bf-cc38-4631-83b1-de6b76a17d27 "nyu-wishlist-service-fall2103") deployed [wishlists](https://github.com/devops-wishlist-buddies/wishlists/releases/tag/deploy-ibm-yp-us-south-nyu-wishlist-service-fall2103-prod-20211206-185611 "deploy-ibm-yp-us-south-nyu-wishlist-service-fall2103-prod-20211206-185611") to [prod](https://console.bluemix.net/apps/634ee8f4-02d9-43db-87bc-85b1a686b8ec?env_id=ibm:yp:us-south "ibm:yp:us-south:nyu-wishlist-service-fall2103:prod"), including commit 535129f62952eba9ba46252344cce9bcc546e5a0

**Pull request merged**